// @vue-pdf-viewer/viewer/dist/annotation-ink-Dq1T0COO.js
import { openBlock as t, createElementBlock as a, createElementVNode as e } from "vue";
const n = {
  xmlns: "http://www.w3.org/2000/svg",
  width: "1em",
  height: "1em",
  fill: "none",
  viewBox: "0 0 20 20"
};
function o(r, l) {
  return t(), a("svg", n, l[0] || (l[0] = [
    e("g", { "clip-path": "url(#a)" }, [
      e("path", {
        fill: "currentColor",
        d: "M16.29 15.038a.83.83 0 0 0 0 1.172.94.94 0 1 1-1.329 1.328 2.704 2.704 0 0 1 0-3.828l.732-.732a.831.831 0 0 0-1.172-1.172l-5.215 5.215a2.707 2.707 0 0 1-3.829-3.828l7.716-7.715a.83.83 0 0 0-1.172-1.172L6.806 9.52a2.707 2.707 0 0 1-3.829-3.828l3.235-3.23A.94.94 0 1 1 7.54 3.79L4.305 7.02a.83.83 0 0 0 1.172 1.172l5.215-5.215a2.707 2.707 0 0 1 3.828 3.829L6.806 14.52a.83.83 0 0 0 1.171 1.172l5.215-5.215a2.708 2.708 0 0 1 3.828 3.829z"
      })
    ], -1),
    e("defs", null, [
      e("clipPath", { id: "a" }, [
        e("path", {
          fill: "#fff",
          d: "M0 0h20v20H0z"
        })
      ])
    ], -1)
  ]));
}
const d = { render: o };
export {
  d as default,
  o as render
};
