// @vue-pdf-viewer/viewer/dist/annotation-highlight-DfO5AU26.js
import { openBlock as l, createElementBlock as t, createElementVNode as n } from "vue";
const o = {
  xmlns: "http://www.w3.org/2000/svg",
  width: "1em",
  height: "1em",
  fill: "none",
  viewBox: "0 0 20 20"
};
function r(a, e) {
  return l(), t("svg", o, e[0] || (e[0] = [
    n("path", {
      fill: "currentColor",
      "fill-rule": "evenodd",
      d: "M12.429 2.033a.617.617 0 0 0-.873 0L8.07 5.518a.617.617 0 0 0 0 .873l6.39 6.39a.617.617 0 0 0 .874 0l3.485-3.484a.617.617 0 0 0 0-.873zm-4.942 5.23-.87.87a.62.62 0 0 0-.152.625c.017.146.06.558.015.841-.064.405-.199.53-.622.922-.16.148-.362.335-.616.59a.617.617 0 0 0 0 .872l3.626 3.628a.617.617 0 0 0 .873 0c.256-.256.444-.459.593-.62.39-.42.514-.555.917-.619.283-.045.696-.002.842.015.22.07.46.011.624-.152l.871-.87zm-5.388 8.562 2.812-2.877 2.88 2.881-1.646 1.61a.62.62 0 0 1-.425.17s-2.974-.704-3.179-.737-.472-.148-.569-.377a.62.62 0 0 1 .127-.67",
      "clip-rule": "evenodd"
    }, null, -1)
  ]));
}
const d = { render: r };
export {
  d as default,
  r as render
};
