// @vue-pdf-viewer/viewer/dist/fullscreen-BmHQQPIs.js
import { openBlock as a, createElementBlock as l, createElementVNode as t } from "vue";
const n = {
  xmlns: "http://www.w3.org/2000/svg",
  width: "1em",
  height: "1em",
  fill: "none",
  viewBox: "0 0 20 20"
};
function o(r, e) {
  return a(), l("svg", n, e[0] || (e[0] = [
    t("path", {
      fill: "currentColor",
      d: "M16.875 3.75V7.5a.625.625 0 1 1-1.25 0V5.259l-3.308 3.308a.625.625 0 0 1-.884-.884l3.308-3.308H12.5a.625.625 0 1 1 0-1.25h3.75a.625.625 0 0 1 .625.625m-9.192 7.683L4.375 14.74V12.5a.625.625 0 1 0-1.25 0v3.75a.625.625 0 0 0 .625.625H7.5a.625.625 0 1 0 0-1.25H5.259l3.308-3.308a.625.625 0 0 0-.884-.884m8.567.442a.624.624 0 0 0-.625.625v2.241l-3.308-3.308a.625.625 0 0 0-.884.884l3.308 3.308H12.5a.624.624 0 1 0 0 1.25h3.75a.624.624 0 0 0 .625-.625V12.5a.624.624 0 0 0-.625-.625m-10.991-7.5H7.5a.625.625 0 0 0 0-1.25H3.75a.625.625 0 0 0-.625.625V7.5a.625.625 0 0 0 1.25 0V5.259l3.308 3.308a.625.625 0 0 0 .884-.884z"
    }, null, -1)
  ]));
}
const c = { render: o };
export {
  c as default,
  o as render
};
